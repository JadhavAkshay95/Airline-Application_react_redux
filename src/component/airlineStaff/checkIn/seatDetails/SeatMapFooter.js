import React, { useState } from 'react';
import './seat.scss';
import Button from 'react-bootstrap/Button';
import PropTypes from 'prop-types';

const SeatMapFooter = ({
  handleSeatOperation,
  selectedPassenger,
  seatChangeOperation,
  enableSeats
}) => {
  const [showSeatChangeMessage, setShowSeatChangeMessage] = useState(false);

  const showMessage = () => {
    if (showSeatChangeMessage) {
      seatChangeOperation();
    } else {
      setShowSeatChangeMessage(true);
      enableSeats(true);
    }
  };
  return (
    <div className="seatMapFooter">
      {showSeatChangeMessage && (
        <h3>
          Select the seat you want change and then click on seat change button
        </h3>
      )}
      {selectedPassenger &&
        !selectedPassenger.checkIn &&
        !selectedPassenger.checkOut && (
          <>
            <Button
              onClick={handleSeatOperation}
              value="checkIn"
              variant="primary"
            >
              Check In
            </Button>
            <Button variant="dark" value="seatChange" onClick={showMessage}>
              Seat change
            </Button>
          </>
        )}
      {selectedPassenger &&
        selectedPassenger.checkIn &&
        !selectedPassenger.checkOut && (
          <>
            <Button
              onClick={handleSeatOperation}
              value="undoCheckIn"
              variant="warning"
            >
              Undo checkIn
            </Button>
            <Button
              variant="success"
              onClick={handleSeatOperation}
              value="checkOut"
            >
              Check out
            </Button>
            <Button variant="dark" value="seatChange" onClick={showMessage}>
              Seat change
            </Button>
          </>
        )}
    </div>
  );
};

export default SeatMapFooter;
SeatMapFooter.propTypes = {
  handleSeatOperation: PropTypes.func,
  selectedPassenger: PropTypes.object,
  seatChangeOperation: PropTypes.func,
  enableSeats: PropTypes.func
};
