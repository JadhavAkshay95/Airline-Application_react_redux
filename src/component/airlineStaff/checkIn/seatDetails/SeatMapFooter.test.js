import React from 'react'
import { shallow } from 'enzyme'
import SeatMapFooter from './SeatMapFooter'

const renderSeatMapFooter = args => {
  const defaultProps = {
    handleSeatOperation: () => {},
    selectedPassenger: () => {},
    seatChangeOperation: () => {}
  }

  const props = { ...defaultProps, ...args }
  return shallow(<SeatMapFooter {...props} />)
}

describe('Render Seat map Footer component', () => {
  it('Render buttons initially', () => {
    const selectedPassenger = {
      checkIn: false,
      checkOut: false
    }
    const wrapper = renderSeatMapFooter({ selectedPassenger })
    expect(wrapper.find('Button')).toHaveLength(2)
  })

  it('Render buttons for check-in status', () => {
    const selectedPassenger = {
      checkIn: true,
      checkOut: false
    }
    const wrapper = renderSeatMapFooter({ selectedPassenger })
    expect(wrapper).toMatchSnapshot()
  })
})
