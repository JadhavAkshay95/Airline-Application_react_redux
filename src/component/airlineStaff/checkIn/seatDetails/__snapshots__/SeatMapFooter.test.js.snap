// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Seat map Footer component Render buttons for check-in status 1`] = `ShallowWrapper {}`;
